spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/BlasWrapper.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.mkldnn
+
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.mkl.Memory
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity, TensorModule}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{MultiShape, Shape}
+import spire.syntax.module
+
+/**
+ * wrap blas module to be dnn module,
+ * and the module should have implemented "computeOutputShape" func.
+ * @param module
+ */
+private[bigdl] class BlasWrapper(val module: AbstractModule[Activity, Activity, Float])
+  extends MklDnnLayer {
+
+  require(!module.isInstanceOf[MklDnnModule], "Only support wrapper blas layer to dnn layer")
+
+  output = module.output
+  gradInput = module.gradInput
+
+  private def inferFormats(inputs: Array[MemoryData]): Int = {
+    // remind: here assume all shapes in inputs should be same
+    inputs.foreach(in =>
+      require(in.shape.length == 2 || in.shape.length == 4,
+      s"just support input shape dim is 2 or 4, but get ${in.shape.length}"))
+
+    inputs(0).layout match {
+      case Memory.Format.nhwc => Memory.Format.nhwc
+      case Memory.Format.nc => Memory.Format.nc
+      case _ => Memory.Format.nchw
+    }
+  }
+
+  override private[mkldnn] def initFwdPrimitives(inputs: Array[MemoryData], phase: Phase) = {
+    // todo: only support model having implemented computeOutputShape
+    val inputShape = inputs.map(in => Shape(in.shape))
+    val outputShape = if (inputShape.length == 1) {
+      // single shape
```

Comment:
Do we need this comment ?  the if condition will tell us obviously right ?

Reply:
ok, just a reminder.